site/guides/sponsoring-gas/gas-sponsorship-eligibility.md

```diff
@@ -0,0 +1,119 @@
+
+But what happens when the user operation you are sending fails to satisfy the criteria set in the gas manager policy? How do you check if the user operation is eligible for gas sponsorship before sending the user operation?
+
+If you do send the user operation that is not eligible for the gas sponsorship under your Gas Manager policy, `buildUserOperation` and `sendUserOperation` will fail due to the error thrown during the [`PaymasterMiddleware`](/packages/aa-core/provider/withPaymasterMiddleware.md) failure. You can follow the guide below to check for gas sponsorship eligibility in advance.
+
+## 1. How to Check for Gas Sponsorship Eligibility for a User Operation
+
+First, you can follow the same instructions from the previous guide on [How to Sponsor Gas for a User Operation](./sponsoring-gas.md) to set up your `SmartAccountProvider` and link your gas policy.
+
+::: code-group
+
+```ts [setup-alchemy-provider.ts]
+import { provider } from "./provider.ts";
+
+// Find your Gas Manager policy id at: // [!code focus:10]
+//dashboard.alchemy.com/gas-manager/policy/create
+const GAS_MANAGER_POLICY_ID = "YourGasManagerPolicyId";
+
+// Link the provider with the Gas Manager. This ensures user operations
+// sent with this provider get sponsorship from the Gas Manager.
+provider.withAlchemyGasManager({
+  policyId: GAS_MANAGER_POLICY_ID,
+});
+```
+
+<<< @/snippets/provider.ts
+
+:::
+
+Then, before you call `sendUserOperation` on the provider, you can use [`buildUserOperation`](/packages/aa-core/provider/buildUserOperation.md) to check if the user operation is eligible for gas sponsorship or not.
+
+Calling `buildUserOperation` method on the provider that is configured with the gas manager (`withAlchemyGasManager`) will run the middleware pipeline including the paymaster middleware, which will thrown an error if the user operation is not eligible for gas sponsorship under the policy.
+
+```ts [check-gas-sponsorship-eligibility.ts]
+let elligible = true;
+
+try {
+  await provider.buildUserOperation({
+    target: "0xTargetAddress",
+    data: "0xCallData",
+    value: 0n, // value in bigint or leave undefined
+  });
+} catch (e) {
+  // if the user operation is not eligible for gas sponsorship, error will be thrown and caught here
+  elligible = false;
+}
+
+console.log(
+  `User Operation is ${
+    elligible ? "elligible" : "inelligible"
+  } for gas sponsorship`
+);
+```
+
+## 2. How to Bypass the Paymaster Middleware For User Operations Not Eligible for Gas Sponsorship
+
+So how would you handle the user operation not elligible for gas sponsorship? You might want to unblock your users to be able to make the user operation transaction without gas sponsorship. This section of the guide explains how you can bypass the `PaymasterMiddleware` and still be able to send the user operation without gas sponsorship. This will let your users to make the transaction by paying the gas fee themselves.
```

Comment:
Pushing back here for some edits. I think it's a dangerous recommendation to suggest overriding the paymasterAndData field without re-asking the user and alerting them they would need to pay for gas themselves: a) because they may not have funds and it'll fail anyways, and b) because even if they had the funds, they might be blind to this.

In addition to the copy changes below, let's add a `tip` section warning developers about re-sending the UO without sponsored paymasterAndData without getting user input first on the dapp as part of good UX. This is per our discussion [here](https://alchemyinsights.slack.com/archives/C05SX4YLJJZ/p1699453433788839?thread_ts=1696611193.112559&cid=C05SX4YLJJZ).

```suggestion
So how would you handle the user operation not eligible for gas sponsorship? You can choose to surface this to your users by catching the error, especially if your app promises no gas fees. You might then want to unblock your users by sending the user operation without gas sponsorship, and having them pay for the gas themselves after giving their permission on your app.

This section of the guide explains how you can bypass the `PaymasterMiddleware` and still be able to send the user operation without gas sponsorship. This will let you send UserOperations where users pay the gas fee themselves via their smart account.
```